codec/gen.go

```diff
@@ -169,7 +173,39 @@ func Gen(w io.Writer, buildTags, pkgName, uid string, useUnsafe bool, ti *TypeIn
 	typ2 := make([]reflect.Type, 0, len(typ))
 	for _, t := range typ {
 		if reflect.PtrTo(t).Implements(selferTyp) || t.Implements(selferTyp) {
-			continue
+			pkg, err := build.Default.Import(t.PkgPath(), "", 0)
```

Comment:
Of course all this code should not go here into `Gen` (which is executed for every type). Parsing once is clearly enough for each package.